presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTaskFactory.java

```diff
@@ -101,6 +88,8 @@ public class HttpRemoteTaskFactory
     private final MetadataManager metadataManager;
     private final QueryManager queryManager;
     private final DecayCounter taskUpdateRequestSize;
+    private final EventLoopGroup eventLoopGroup = new DefaultEventLoopGroup(Runtime.getRuntime().availableProcessors(),
+            new ThreadFactoryBuilder().setNameFormat("task-event-loop-%s").setDaemon(true).build());
```

Comment:
Can you explain the reasoning for choosing `DefaultEventLoopGroup` instead of other implementations of `EventLoopGroup`?

```suggestion
    private final EventLoopGroup eventLoopGroup = new DefaultEventLoopGroup(
            Runtime.getRuntime().availableProcessors(),
            new ThreadFactoryBuilder().setNameFormat("task-event-loop-%s").setDaemon(true).build());
```

Reply:
Absolutely. Netty provides a few event loop group implementations. A quick comparison among them could be: 

- NioEventLoopGroup: Most commonly used, best for TCP/IP socket communications
- EpollEventLoopGroup: Linux-specific implementation, better performance than NIO on Linux systems
- KQueueEventLoopGroup: MacOS/BSD-specific implementation, better performance on MacOS/BSD systems
- DefaultEventLoopGroup: For non-network operations, used for local event processing
- LocalEventLoopGroup: Optimized for local transport (in-VM pipe communication)

In our case, the network async calls are already being handled by http client and this event loop is primarily being used for task scheduling and coordination (method calls, state management, etc.). So defaultEventLoopGroup might be a better choice.  Please let me know if this makes sense and thanks a lot for the comments!